### ISSUE TITLE
FilterSortBottomSheet taking a long time to open

### ISSUE THREAD

**Issue by FineFindus** [libre-tube/LibreTube#5650]
### Steps to reproduce

1. Open FilterSortBottomSheet

### Expected behavior

The sheet should open within a reasonable timeframe.

### Actual behavior

The sheet takes a while. The device is under heavy load, dropping frames.

### LibreTube version

bfa55d327868e58bf1fe22cc2fd2c86f3f62f370

### Android version

Android 13

### Other details

This appears to only happen when the users have a larger amount of subscription loaded. Maybe it's related to a low amount of available memory?
From my testing, it seems like the [`setInitialFiltersStae()`](https://github.com/libre-tube/LibreTube/blob/9b963a3e28471d5a88045dba5f66485e681b390c/app/src/main/java/com/github/libretube/ui/sheets/FilterSortBottomSheet.kt#L84) function takes the longest to execute.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
We had set up the listeners before setting the initial state, so the whole feed was re-generated each time we have programmatically toggled a filter option when loading the defaults / latest state. So the feed (recycler view) content got re-created about 5 times in parallel only when opening the sheet ...
